src/StateSmith/Output/Sim/HtmlRenderer.cs

```diff
@@ -358,14 +359,12 @@ public static void Render(StringBuilder stringBuilder, string smName, string moc
         // when using {{smName}}.js in your own applications, although you may
         // choose to implement a tracer for debugging purposes.
         sm.tracer = {
-            enterState: (stateId) => {
-                var name = {{smName}}.stateIdToString(stateId);
-                document.querySelector('g[data-id=' + name + ']')?.classList.add('active');
-                sm.tracer.log(""Entered "" + name);
+            enterState: (mermaidName) => {
```

Comment:
oh interesting, looks like you ran into the issue in my previous comment earlier than i expected!

the tracer probably shouldn't be aware of mermaid names. It should only know the states by the IDs in the SM. Maybe the right way to do this is to have the tracer.enterState() take a stateId, and then provide a mapping function in the html that maps stateIds to mermaid names?

If we want to leave it as-is for now that's fine, we'll just need to migrate it once we do #278

Reply:
I agree a mapping function would feel cleaner, but I'd prefer to leave it for now in preference of working on diagram-only.

I'm not sure we will end up using the future built in tracer for the sim. We already use more features than planned for the built in. I guess we probably could if we extend the auto generated built in tracer. We'll have a better idea when we get closer.